### PR TITLE
Added ItemsAdder Block Support

### PR DESCRIPTION
**Pull request by Teenkung123**
In this request, I added support for ItemsAdder block generation by specifying the ItemsAdder custom block name, space, ID, and probability like this:

image

(It's my first time opening a pull request, so if there's anything I need to fix, please let me know.)

### CODE REVIEW

**Comment by HSGamer** [comment]
Check out the `OneBlockCustomBlockCreator` and #283 

I've added a creator so that you can just make an addon to register your `OneBlockCustomBlock`, with the catch that the format for `blocks` is different.

**Comment by Teenkung123** [comment]
I will try doing that, but I'm not sure if it will work because I have to place ItemsAdder's custom block using customBlock.place(); otherwise, it won't recognize it and will think it's just a normal noteblock with the wrong note.

https://itemsadder.devs.beer/developers/java-api/examples

```java
CustomBlock customBlock = CustomBlock.getInstance("ruby_ore");
if(customBlock != null) //not needed if you're sure the blocks exists.
{
    customBlock.place(location);
}
else
{
    // Custom block not found in ItemsAdder configurations!
}
```

**Comment by HSGamer** [comment]
You can use `block.getLocation()` to get the location and use that method.
Just make sure to check the custom block when registering on the creator (return empty if it's not found)

**Comment by Teenkung123** [comment]
I made it check on load whether the custom blocks exist in ItemsAdder or not. If they don't exist, it will log an error to the console saying the custom block does not exist. If they do exist, it will add the NamespaceID to OneBlockObject.

```java
            for (String material : blocks.getKeys(false)) {
                if (Material.getMaterial(material) != null) {
                    addMaterial(obPhase, material, Objects.toString(blocks.get(material)));
                } else {
                    if (addon.hasItemsAdder()) {
                        CustomBlock block = CustomBlock.getInstance(material);
                        if (block != null) {
                            addItemsAdderBlock(obPhase, material, Objects.toString(blocks.get(material)));
                        } else {
                            addon.logError("Bad block material in " + obPhase.getPhaseName() + ": " + material);
                        }
                    } else {
                        addon.logError("Bad block material in " + obPhase.getPhaseName() + ": " + material);
                    }
                }
            }
```

In the OneBlockObject class, I added a String variable that will store the NamespaceID of the CustomBlock. 
In the BlockListener, I added a secondary check to ensure that the blocks really exist before placing the CustomBlock.

```java
        if (nextBlock.isCustomBlock()) {
            nextBlock.getCustomBlock().setBlock(block);
            return;
        } else if (nextBlock.isItemsAdderBlock()) {
            //Get Custom Block from ItemsAdder and place it
            CustomBlock cBlock = CustomBlock.getInstance(nextBlock.getItemsAdderBlock());
            if (cBlock != null) {
                cBlock.place(block.getLocation());
                return;
            }
        }
```

File: src/main/java/world/bentobox/aoneblock/AOneBlock.java [hunk 1]
```diff
@@ -34,22 +35,28 @@
...
     private static final String THE_END = "_the_end";
+    public static boolean hasItemsAdder = false;
 
     // Settings
     private Settings settings;
     private ChunkGeneratorWorld chunkGenerator;
     private final Config<Settings> configObject = new Config<>(this, Settings.class);
-    private BlockListener blockListener;
-    private OneBlocksManager oneBlockManager;
+    private static BlockListener blockListener;
```
Comment (tastybento) [inline src/main/java/world/bentobox/aoneblock/AOneBlock.java:48]: Why did this become static?

File: src/main/java/world/bentobox/aoneblock/AOneBlock.java [hunk 2]
```diff
@@ -34,22 +35,28 @@
...
+    private static OneBlocksManager oneBlockManager;
```
Comment (tastybento) [inline src/main/java/world/bentobox/aoneblock/AOneBlock.java:49]: Why did this become static?

File: src/main/java/world/bentobox/aoneblock/AOneBlock.java [hunk 3]
```diff
@@ -79,18 +86,8 @@ private boolean loadSettings() {
...
-            blockListener = new BlockListener(this);
-        } catch (IOException e) {
-            // Disable
-            logError("AOneBlock settings could not load (oneblock.yml error)! Addon disabled.");
-            logError(e.getMessage());
-            setState(State.DISABLED);
-            return;
-        }
-        registerListener(blockListener);
+        loadData(true);
```
Comment (tastybento) [inline src/main/java/world/bentobox/aoneblock/AOneBlock.java:89]: This parameter is hardcoded true. Why is it needed?

File: src/main/java/world/bentobox/aoneblock/AOneBlock.java [hunk 4]
```diff
@@ -106,6 +103,24 @@ public void onEnable() {
         registerListener(holoListener);
     }
 
+    //Load some of Manager
+    public void loadData(boolean register) {
```
Comment (tastybento) [inline src/main/java/world/bentobox/aoneblock/AOneBlock.java:107]: I see this is called by src/main/java/world/bentobox/aoneblock/listeners/ItemsAdderListener.java but not sure why it needs the parameter.

File: src/main/java/world/bentobox/aoneblock/listeners/ItemsAdderListener.java [hunk 5]
```diff
@@ -0,0 +1,20 @@
+package world.bentobox.aoneblock.listeners;
+
+import dev.lone.itemsadder.api.Events.ItemsAdderLoadDataEvent;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import world.bentobox.aoneblock.AOneBlock;
+
+public class ItemsAdderListener implements Listener {
```
Comment (tastybento) [inline src/main/java/world/bentobox/aoneblock/listeners/ItemsAdderListener.java:8]: Add some JavaDoc to explain what this class is for please.

**Review by tastybento — CHANGES_REQUESTED** [review]
Please see the review comments. Thanks for the PR.

**Comment by Teenkung123** [comment]
Thanks for the review. I have resolved all of the issues in the [following commit](https://github.com/BentoBoxWorld/AOneBlock/pull/317/commits/d63a744d85fef1150798f2dd9a6ccde00d5d0922).

**Review by tastybento — APPROVED** [review]

**Pull request merged**